Sprint22 Задачи

AlexanderKorytin opened the pull request:
Поправил верстку. курсоры сразу не заметил. Сейчас сделал прозрачными как в Figme. Баг с повторным поиском не исправил, т.к. придется переписывать логику работы с избранными треками, а именно переносить проверку избранности из поискового фрагмента в плеер. Я задавал вопрос в пачке - рекомендовали при поиске! А без этого будет неправильно отображаться значек избранности при повторном заходе в плеер. За ToolBar спасибо, пока изучаю...

AlexanderKorytin committed:
- обновление RecyclerView's реализовано с использованием DiffUtil

AlexanderKorytin committed:
- рефакторинг

AlexanderKorytin committed:
- в DiffUtil CallBack сменен на ItemCallback<TrackUI>()

AlexanderKorytin committed:
- рефакторинг

AlexanderKorytin committed:
Sprint 21
- click по элементу списка теперь выполняется корректно(после добавления в Адаптер trackClickListner.onTrackClick(currentList[holder.adapterPosition])

AlexanderKorytin committed:
- сделана верстка фрагмента создания нового плейлиста

AlexanderKorytin committed:
Sprint 22
- добавлен RecyclerView в верстку фраглента альбомов
- добавлены обработчики кнопок - создать альбом( фрагмент альбомов) и назад(фрагмент создания альбома)
- реализован permission Camera
- реализован выбор картинки альбома и сохранение во внутреннее хранилище

AlexanderKorytin committed:
Sprint 22
- рефакторинг

AlexanderKorytin committed:
Sprint 22
- написана база данных плейлистов, репозиторий и интерактор создания плейлистов
- реализована вьюмодель для создания плейлистов
- написан диалог для выхода из создания плейлистов при зополнеон поле названия или описания или выбранной каринке,
- выбранная обложка сохраняется во внутреннее хранилище

AlexanderKorytin committed:
Sprint 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- сделана верстка playlist_item.xml
- доделана верстка tab_albums_fragment.xml
- написан PlayListViewHolder.kt

AlexanderKorytin committed:
спринт 22
- написан PlayListsDiffUtil.kt
- написан PlayListAdapter.kt

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- рефакторинг
- написана моделька PlayListsScreenState.kt
- изменена логика сохранения названия файла изображения обложки в базу данных
- реализована логика выгрузки картинки обложки из внутреннего хранилища

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- рефакторинг
- конуц задачи 2

AlexanderKorytin committed:
спринт 22
- написана функция написания слова трек в графе колличества плейлиста

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- сделана верстка всплывающего диалога в плеере

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- доделана верстка экрана плеера под диалог
-сделана верстка элемента списка плейлистов под диалог

AlexanderKorytin committed:
спринт 22
- рефакторинг

AlexanderKorytin committed:
спринт 22
- написаны PlayListPlayerViewHolder, PlaylistsDiffUtilPlayer, PlayListsPlayerAdapter

AlexanderKorytin committed:
Sprint 22
- в PlatListsPlayerAdapter добавлен интерфейс для обработки нажатия

AlexanderKorytin committed:
Sprint 22
- медиаплеер переведен на Fragment
- изменен слушатель системной кнопки назад на экрвне создания плей листов
- реализован показ списка плейлистов на выезжаюшем листе медиа плеера
- добавлены анимации переходов

AlexanderKorytin committed:
Sprint 22
- дополнен PlayListDbConverter, AppDataBase
- написан AllTracksDao, TrackPLEntity

AlexanderKorytin committed:
Sprint 22
- дописан AllTracksDao
- реализовано сохранение трека в базу
- обновление плейлиста пока не работает

AlexanderKorytin committed:
Sprint 22
- обновление плейлиста работает

AlexanderKorytin committed:
Sprint 22
- исправлена ошибочная работаб когда создавался только один лист и далее перезаписывался
- колличество треков в плейлисте на экране плеера обномляется
- исправлен баг некоректного отображения состояния экрана в части всплывающего окна при возврате из экрана создания листа

AlexanderKorytin committed:
Sprint 22
- исправлен цвет фона экрана создания альбома

IliaAnK commented on app/src/main/res/layout/album_creator_fragment.xml:
```diff
@@ -0,0 +1,99 @@
+            android:background="@drawable/new_playlist_button"
+            android:enabled="false"
+            android:fontFamily="@font/ys_display_regular"
+            android:paddingHorizontal="12.5dp"
```
Рекомендация:
1. Дробные числа не используются для размеров;
2. Этот атрибут ни на что не влияет. Можно просто удалить

IliaAnK commented on app/src/main/res/layout/album_creator_fragment.xml:
```diff
@@ -0,0 +1,99 @@
+            android:id="@+id/create_playlist"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginHorizontal="17dp"
```
Рекомендация:
В дизайне могут быть нечетные размеры, или не симметричные. Но лучше использовать в коде размеры, кратные 4 или 8. 
Документация - https://material.io/design/layout/spacing-methods.html#baseline-grid

AlexanderKorytin committed:
Sprint 22
- исправлен фон BottonNavBar
- к BottonNavBar добавлена разделительная полоса сверху
- изменена верстка album_creator_fragment, что бы кнопка Создать всегда была на экране

AlexanderKorytin committed:
Sprint 22
- нв экрне создания альбомов клавиатура не больше перекрывает поля ввода

AlexanderKorytin committed:
Sprint 22
- нв экрне создания альбомов клавиатура не больше перекрывает поля ввода

AlexanderKorytin committed:
Sprint 22
- увеличен зазор между столбцами списка плейтистов до 8dp

AlexanderKorytin committed:
Sprint 22
- объединил favorite и favoritetracks
- поля ввода текста экрана создания плейлиста переделан в соответсвии с версткой

AlexanderKorytin committed:
Sprint 22
- добавлены переводы текста alert dialog экрана создания плейлистов
- цвет кнопок alert dialog экрана создания плейлистов изменен

AlexanderKorytin committed:
Sprint 22
- изменена логика изменения цвета рамок EditText и появления текста на них экрана создания альбома, добавлен слушатель изменния фокуса

AlexanderKorytin committed:
Sprint 22
- рефакторинг

AlexanderKorytin committed:
Sprint 22
- изменена livedata показа тоста добавления трека в базу, что бы при показе тоста и  переходе на другой экран и обратно тост не показывался снова

AlexanderKorytin committed:
Sprint 22
- рефакторинг

IliaAnK commented:
Макет:
<img width="344" alt="Снимок экрана 2023-12-11 в 8 03 17 PM" src="https://github.com/AlexanderKorytin/PlaylistMaker/assets/135727674/e81d1336-9bf7-442b-ae43-f3a2b9b7c010">


Приложение:
<img width="342" alt="Снимок экрана 2023-12-11 в 8 03 40 PM" src="https://github.com/AlexanderKorytin/PlaylistMaker/assets/135727674/4f8ffd94-2e2c-49b2-8aa7-eaa3a7d3c069">

AlexanderKorytin committed:
Sprint 22
- На экране "Медиатека", таб "Плейлисты" у заглушек обложки убран фон и размер "змейки" уменьшен в соответсвии с макетом

AlexanderKorytin committed:
Sprint 22
- На экране "Новый плейлист" цвет курсора в полях ввода изменен на безцветный как на макете

AlexanderKorytin committed:
Sprint 22
- убрана лишняя вложенность

AlexanderKorytin committed:
Sprint 22
- настройки темной темы при первом запуске приложения раз читаются из системы

AlexanderKorytin committed:
спринт 22
- На экране "Новый плейлист" цвет курсора в полях ввода изменен на такойже как во фрагменте поиск
